### PR TITLE
fix(VIPEROMCT-412): return a resolved `Promise`

### PR DESCRIPTION
**Pull request by ozyx**
Closes VIPEROMCT-412

### Describe your changes:

Returns a resolved `Promise` so that previous consumers who are expecting a `Promise` logic will not break, or new consumers can use the value directly.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/providers/limit-provider.js [hunk 1]
```diff
@@ -100,9 +100,7 @@ export default class LimitProvider {
         const limits = domainObject.configuration.limits;
 
         return {
-            limits: () => {
-                return limits;
-            }
+            limits: () => Promise.resolve(limits)
```
Comment (scottbell) [inline src/providers/limit-provider.js:103]: Nitpicky, but can we just do:
```javascript

limits: async () => { return limits; }
```

Comment (scottbell) [reply src/providers/limit-provider.js:103]: Probably will complain it’s missing an await! Ignore me

Comment (ozyx) [reply src/providers/limit-provider.js:103]: nitpicky but neat! I like it

**Comment by khalidadil** [comment]
Verified Fixed in Testathon on 09/22/23 under "Gauges do not work with yamcs telemetry" in Viper Jira

**Pull request merged**